DayTomato/DayTomato.Droid/Adapters/ViewPinAdapter.cs

```diff
@@ -0,0 +1,147 @@
+				{
+					_pinLiked[position] = false;
+					_pinDisliked[position] = false;
+					_pins[position].Likes++;
+					vh.PinLikes.Text = _pins[position].Likes.ToString();
+					vh.UpButton.SetImageResource(Resource.Drawable.up_arrow_unfilled);
+					vh.DownButton.SetImageResource(Resource.Drawable.down_arrow_unfilled);
+				}
+			};
+			vh.DownButton.Click += (sender, e) =>
+			{
+				// If the like and dislike button was not pressed, then its fresh
+				if (!_pinLiked[position] && !_pinDisliked[position])
+				{
+					_pinLiked[position] = false;
+					_pinDisliked[position] = true;
+					_pins[position].Likes--;
+					vh.PinLikes.Text = _pins[position].Likes.ToString();
+					vh.UpButton.SetImageResource(Resource.Drawable.up_arrow_unfilled);
+					vh.DownButton.SetImageResource(Resource.Drawable.down_arrow_filled);
+				}
+				// Else we need to "reset" the likes
+				else if (_pinLiked[position])
+				{
+					_pinLiked[position] = false;
+					_pinDisliked[position] = false;
+					_pins[position].Likes--;
+					vh.PinLikes.Text = _pins[position].Likes.ToString();
+					vh.UpButton.SetImageResource(Resource.Drawable.up_arrow_unfilled);
+					vh.DownButton.SetImageResource(Resource.Drawable.down_arrow_unfilled);
+				}
+			};
+		}
+	}
+
+	public class ViewHolder : RecyclerView.ViewHolder
+	{
+		private Action<int> _listener;
+
+		public ImageView PinImage { get; private set; }
+		public TextView PinName { get; private set; }
+		public ImageView UpButton { get; private set; }
+		public TextView PinLikes { get; private set; }
+		public ImageView DownButton { get; private set; }
+		public TextView PinDescription { get; private set; }
+		public TextView PinReview { get; private set; }
+		public TextView PinLinkedAccount { get; private set; }
+
+		public ViewHolder(View itemView) : base (itemView)
+		{
+			PinImage = itemView.FindViewById<ImageView>(Resource.Id.pin_view_holder_pin_image);
+			PinName = itemView.FindViewById<TextView>(Resource.Id.pin_view_holder_pin_name);
+			UpButton = itemView.FindViewById<ImageView>(Resource.Id.pin_view_holder_up_button);
+			PinLikes = itemView.FindViewById<TextView>(Resource.Id.pin_view_holder_likes);
+			DownButton = itemView.FindViewById<ImageView>(Resource.Id.pin_view_holder_down_button);
+			PinDescription = itemView.FindViewById<TextView>(Resource.Id.pin_view_holder_description);
+			PinReview = itemView.FindViewById<TextView>(Resource.Id.pin_view_holder_review);
+			PinLinkedAccount = itemView.FindViewById<TextView>(Resource.Id.pin_view_holder_account);
+		}
+
+		public void SetClickListener(Action<int> listener)
+		{
+			_listener = listener;
+			ItemView.Click += HandleClick;
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			base.Dispose(disposing);
+			if (ItemView != null)
+			{
+				ItemView.Click -= HandleClick;
+			}
+			_listener = null;
+		}
+
+		void HandleClick(object sender, EventArgs e)
+		{
+			if (_listener != null)
+			{
+				_listener(base.AdapterPosition);
```

Comment:
It may be possible to remove the conditional and simply do `_listener?.Invoke(..)` :)